src/include/storage/sql_table.h

```diff
@@ -71,8 +72,8 @@ class SqlTable {
    * @param redo the desired change to be applied. This should be the after-image of the attributes of interest.
    * @return true if successful, false otherwise
    */
-  bool Update(transaction::TransactionContext *const txn, const TupleSlot slot, const ProjectedRow &redo) const {
-    return table_.data_table->Update(txn, slot, redo);
+  bool Update(transaction::TransactionContext *const txn, const TupleSlot slot, RedoRecord *const redo) const {
```

Comment:
Should this take `TupleSlot` still or should it read it from the redo record as it passes it down to the `DataTable`?  I think that would semantically cleaner.

Reply:
Confirmed with @tli2 and we're in agreement that we can lose it as an argument. I'll make some TERRIER_ASSERTs to make sure it's populated.